cmd/oras/root/manifest/index/update.go

```diff
@@ -0,0 +1,235 @@
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.RawReference = args[0]
+			return option.Parse(cmd, &opts)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return updateIndex(cmd, opts)
+		},
+	}
+	option.ApplyFlags(&opts, cmd.Flags())
+	cmd.Flags().StringArrayVarP(&opts.addArguments, "add", "", nil, "manifests to add to the index")
+	cmd.Flags().StringArrayVarP(&opts.mergeArguments, "merge", "", nil, "indexes to be merged into the index")
+	cmd.Flags().StringArrayVarP(&opts.removeArguments, "remove", "", nil, "manifests to remove from the index, must be digests")
+	cmd.Flags().StringArrayVarP(&opts.tags, "tag", "", nil, "extra tags for the updated index")
+	return oerrors.Command(cmd, &opts.Target)
+}
+
+func updateIndex(cmd *cobra.Command, opts updateOptions) error {
+	// if no update flag is used, do nothing
+	if !updateFlagsUsed(cmd) {
+		opts.Println("No update flag is used. There's nothing to update.")
+		return nil
+	}
+	ctx, logger := command.GetLogger(cmd, &opts.Common)
+	target, err := opts.NewTarget(opts.Common, logger)
+	if err != nil {
+		return err
+	}
+	if err := opts.EnsureReferenceNotEmpty(cmd, true); err != nil {
+		return err
+	}
+	index, err := fetchIndex(ctx, target, opts)
+	if err != nil {
+		return err
+	}
+	manifests, err := removeManifestsFromIndex(ctx, index.Manifests, target, opts)
+	if err != nil {
+		return err
+	}
+	manifests, err = addManifestsToIndex(ctx, manifests, target, opts)
+	if err != nil {
+		return err
+	}
+	manifests, err = mergeIndexes(ctx, manifests, target, opts)
+	if err != nil {
+		return err
+	}
+
+	index.Manifests = manifests
+	indexBytes, err := json.Marshal(index)
+	if err != nil {
+		return err
+	}
+	desc := content.NewDescriptorFromBytes(index.MediaType, indexBytes)
+
+	printUpdateStatus(status.IndexPromptUpdated, string(desc.Digest), "", opts.Printer)
+	path := getPushPath(opts.RawReference, opts.Type, opts.Reference, opts.Path)
+	return pushIndex(ctx, target, desc, indexBytes, opts.Reference, opts.tags, path, opts.Printer)
+}
+
+func fetchIndex(ctx context.Context, target oras.ReadOnlyTarget, opts updateOptions) (ocispec.Index, error) {
+	printUpdateStatus(status.IndexPromptFetching, opts.Reference, "", opts.Printer)
+	desc, content, err := oras.FetchBytes(ctx, target, opts.Reference, oras.DefaultFetchBytesOptions)
+	if err != nil {
+		return ocispec.Index{}, fmt.Errorf("could not find the index %s: %w", opts.Reference, err)
+	}
+	if !descriptor.IsIndex(desc) {
+		return ocispec.Index{}, fmt.Errorf("%s is not an index", opts.Reference)
+	}
+	printUpdateStatus(status.IndexPromptFetched, opts.Reference, string(desc.Digest), opts.Printer)
+	var index ocispec.Index
+	if err := json.Unmarshal(content, &index); err != nil {
+		return ocispec.Index{}, err
+	}
+	return index, nil
+}
+
+func addManifestsToIndex(ctx context.Context, manifests []ocispec.Descriptor, target oras.ReadOnlyTarget, opts updateOptions) ([]ocispec.Descriptor, error) {
```

Comment:
nit: I found it a bit strange to include `ToIndex` in the function name, as the function neither takes an index as a parameter nor return an index🤔

Reply:
Naming is so hard. Renamed the function back to `addManifests`.